test/provideTransactionFeedback.js

```diff
@@ -3,24 +3,170 @@ const InkProtocol = artifacts.require("./mocks/InkProtocolMock.sol")
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Confirmed
+      })
+
+      await $util.assertVMExceptionAsync(protocol.provideTransactionFeedback(transaction.id, rating, comment, { from: unknown }))
+    })
+
+    it("fails when transaction does not exist", async () => {
+      await $util.assertVMExceptionAsync(protocol.provideTransactionFeedback(0, rating, comment))
+    })
+
+    it("fails when transaction state is Revoked", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Revoked
+      })
+
+      await $util.assertVMExceptionAsync(protocol.provideTransactionFeedback(transaction.id, rating, comment, { from: buyer }))
+    })
+
+    it("fails when transaction state is Accepted", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted
+      })
+
+      await $util.assertVMExceptionAsync(protocol.provideTransactionFeedback(transaction.id, rating, comment, { from: buyer }))
+    })
+
+    it("fails when transaction state is Disputed", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Disputed
+      })
+
+      await $util.assertVMExceptionAsync(protocol.provideTransactionFeedback(transaction.id, rating, comment, { from: buyer }))
+    })
+
+    it("fails when transaction state is Escalated", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Escalated
+      })
+
+      await $util.assertVMExceptionAsync(protocol.provideTransactionFeedback(transaction.id, rating, comment, { from: buyer }))
+    })
+
+    it("fails when rating is invalid", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Confirmed
+      })
+
+      let invalidRating = 6
+      await $util.assertVMExceptionAsync(protocol.provideTransactionFeedback(transaction.id, invalidRating, comment, { from: buyer }))
+    })
+
+    it("emits the FeedbackUpdated event", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Confirmed
+      })
+
+      let tx = await protocol.provideTransactionFeedback(transaction.id, rating, comment, { from: buyer })
+      let eventArgs = $util.eventFromTx(tx, $util.events.FeedbackUpdated).args
+
+      assert.equal(eventArgs.transactionId, transaction.id)
+    })
+
+    it("allows multiple calls", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Confirmed
```

Comment:
I saw https://github.com/InkProtocol/ink-protocol/blob/master/contracts/InkProtocol.sol#L521 
It check `require(_transaction.state == TransactionState.Null);`
Why it pass in confirmed state?